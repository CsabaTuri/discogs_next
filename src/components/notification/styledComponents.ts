import Card from "react-bootstrap/Card";
import styled from "styled-components";
export const StyledNotification = styled(Card)`
  position: fixed;
  top: 90%;
  width: 20%;
  right:5%;
  text-align: center;
  z-index: 10;
  opacity:.8;
  .danger {
    background-color: ${(props) => props.theme.danger};
    color: white;
  }
  .success {
    background-color: ${(props) => props.theme.success};
    color: white;
  }
  .warning {
    background-color: ${(props) => props.theme.warning};
    color: white;
  }
`;
