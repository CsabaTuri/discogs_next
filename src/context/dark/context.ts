import { Context as C, createContext, ProviderExoticComponent, useContext } from "react"

let initialState: any = { isDark: true, dispatch: () => { } }

const Context: C<any> = createContext<any>(initialState)

export const Provider: ProviderExoticComponent<any> = Context.Provider;

export const useDarkContext = () => useContext(Context);

export default Context

