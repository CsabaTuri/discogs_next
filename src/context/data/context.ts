import { createContext ,useContext} from "react"

let initialState:any = {  data: {} ,dispatch: () => {}  };

 const Context = createContext(initialState)

 export const useDataContext = () => useContext(Context);

 export default Context
