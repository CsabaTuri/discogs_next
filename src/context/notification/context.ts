import { createContext, useContext } from "react";

let initialState:any = { state: { text: "", type: "success" },dispatch: () => {}  };

const Context = createContext(initialState);

export const useNotificationContext = () => useContext(Context);

export default Context;
