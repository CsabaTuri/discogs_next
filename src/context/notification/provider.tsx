import React, { useReducer } from "react";
import { reducer } from "./reducer";
import Context from "./context";

let initialState = { state: { text: "", type: "success" },dispatch: () => {}  };

export const Provider = ({ children }:any) => {
  const [state, dispatch] = useReducer(reducer, initialState);
  return (
    <Context.Provider value={{ state, dispatch }}>{children}</Context.Provider>
  );
};
