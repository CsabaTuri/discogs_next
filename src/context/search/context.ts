import { createContext ,useContext} from "react"

let initialState:any = { search:{search: {}},dispatch: () => {} }

 const Context = createContext(initialState)

 export const useSearchContext = () => useContext(Context);

 export default Context
