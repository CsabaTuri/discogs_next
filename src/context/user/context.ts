import { Context as C, createContext, ProviderExoticComponent, useContext } from "react"



let initialState: any = { user: {}, dispatch: () => { } };

const Context: C<any> = createContext<any>(initialState)

export const Provider: ProviderExoticComponent<any> = Context.Provider;

export const useUserContext = () => useContext(Context);

export default Context
