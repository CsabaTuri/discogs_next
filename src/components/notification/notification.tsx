import React from "react";
import { StyledNotification } from "./styledComponents";
import { useNotificationContext } from "../../context/notification/context";
const Notification = () => {
  const { state } = useNotificationContext();
  const { text, type } = state;
  console.log(state)
  const [show, setShow] = React.useState(false);
  const [mount, setMount] = React.useState(false);
  React.useEffect(() => {
    if (mount) {
      setShow(true);
      setTimeout(() => {
        setShow(false);
      }, 3000);
    }
    setMount(true);
  }, [state]);
  return (
    <StyledNotification style={{ display: show ? "block" : "none" }}>
      <StyledNotification.Body className={type}>{text}</StyledNotification.Body>
    </StyledNotification>
  );
};
export default Notification;
