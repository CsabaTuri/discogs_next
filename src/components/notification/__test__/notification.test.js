import React from "react";
import { render, screen, mount } from "@testing-library/react";
import renderer from "react-test-renderer";
import Enzyme from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Notification from "../notification";
import * as NotificationContext from "../../../context/notification/context";
import "jest-styled-components";

Enzyme.configure({ adapter: new Adapter() });

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/",
      pathname: "/",
      query: "",
      asPath: "/",
    };
  },
}));
jest.mock("react-i18next", () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));
describe("Notification", () => {
  test("snapshot test", async () => {
    await new Promise((resolve) => {
      window.setTimeout(resolve, 0);
    });

    const contextValues = { state: { type: "success", text: "Success" } };
    jest
      .spyOn(NotificationContext, "useNotificationContext")
      .mockImplementation(() => contextValues);
    const wrapper = Enzyme.mount(<Notification />);
    expect(wrapper).toMatchSnapshot();
  });
});
