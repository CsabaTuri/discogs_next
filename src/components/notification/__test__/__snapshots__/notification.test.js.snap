// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification snapshot test 1`] = `
.c0 {
  position: fixed;
  top: 90%;
  width: 20%;
  right: 5%;
  text-align: center;
  z-index: 10;
  opacity: .8;
}

.c0 .danger {
  color: white;
}

.c0 .success {
  color: white;
}

.c0 .warning {
  color: white;
}

<Notification>
  <styledComponents__StyledNotification
    style={
      Object {
        "display": "none",
      }
    }
  >
    <Card
      body={false}
      className="c0"
      style={
        Object {
          "display": "none",
        }
      }
    >
      <div
        className="c0 card"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <CardBody
          className="success"
        >
          <div
            className="success card-body"
          >
            Success
          </div>
        </CardBody>
      </div>
    </Card>
  </styledComponents__StyledNotification>
</Notification>
`;
